examples/pytorch_lightning_simple.py

```diff
@@ -0,0 +1,190 @@
+
+    def forward(self, data):
+        return self._model(data)
+
+    def training_step(self, batch, batch_nb):
+        data, target = batch
+        output = self.forward(data)
+        loss = F.nll_loss(output, target)
+        return {'loss': loss}
+
+    def validation_step(self, batch, batch_nb):
+        data, target = batch
+        output = self.forward(data)
+        pred = output.argmax(dim=1, keepdim=True)
+        correct = pred.eq(target.view_as(pred)).sum().item()
+        accuracy = correct / data.size(0)
+        return {'validation_accuracy': accuracy}
+
+    def validation_end(self, outputs):
+        accuracy = sum(x['validation_accuracy'] for x in outputs) / len(outputs)
+        # Pass the accuracy to the `DictLogger` via the `'log'` key.
+        return {'log': {'accuracy': accuracy}}
+
+    def configure_optimizers(self):
+        return getattr(optim, self._optimizer_name)(self._model.parameters(), lr=self._lr)
+
+    @pl.data_loader
+    def train_dataloader(self):
+        return torch.utils.data.DataLoader(
+            datasets.MNIST(DIR, train=True, download=True, transform=transforms.ToTensor()),
+            batch_size=BATCHSIZE, shuffle=True)
+
+    @pl.data_loader
+    def val_dataloader(self):
+        return torch.utils.data.DataLoader(
+            datasets.MNIST(DIR, train=False, download=True, transform=transforms.ToTensor()),
+            batch_size=BATCHSIZE, shuffle=False)
+
+
+def objective(trial):
+    # PyTorch Lightning will try to restore model parameters from previous trials if checkpoint
+    # filenames match. Therefore, the filenames for each trial must be made unique.
+    checkpoint_callback = pl.callbacks.ModelCheckpoint(
+        os.path.join(DIR, 'results', 'trial_{}'.format(trial.number)))
+
+    # The default logger in PyTorch Lightining writes to event files to be consumed by
+    # TensorBoard. We create a simple logger instead that holds the log in memory so that the
+    # final accuracy can be obtained after optimization. When using the default logger, the
+    # final accuracy could be stored in an attribute of the `Trainer` instead.
+    logger = DictLogger()
+
+    trainer = pl.Trainer(
+        logger=logger,
+        train_percent_check=PERCENT_TRAIN_EXAMPLES,
+        val_percent_check=PERCENT_TEST_EXAMPLES,
+        checkpoint_callback=checkpoint_callback,
+        max_nb_epochs=EPOCHS,
+        gpus=0 if torch.cuda.is_available() else None,
+    )
+
+    # Generate the model.
+    model = Net(trial)
+
+    # Sample optimizer parameters.
+    optimizer_name = trial.suggest_categorical('optimizer', ['Adam', 'RMSprop', 'SGD'])
+    lr = trial.suggest_uniform('lr', 1e-5, 1e-1)
+
+    # Generate the PyTorch Lightning model.
+    model = LightningNet(model, optimizer_name, lr)
+    trainer.fit(model)
+
+    return logger.metrics[-1]['accuracy']
+
+
+if __name__ == '__main__':
+    study = optuna.create_study(direction='maximize')
+    study.optimize(objective, n_trials=100)
```

Comment:
Recently, we use the `timeout` argument to limit CI time.
```suggestion
    study.optimize(objective, n_trials=100, timeout=600)
```

Reply:
Thanks for the suggestion, applied!